core/templates/components/state-editor/state-editor.directive.ts

```diff
@@ -121,7 +125,19 @@ angular.module('oppia').directive('stateEditor', [
             $scope.conceptCardIsShown = !$scope.conceptCardIsShown;
           };
 
+          var questionSuggestionDiv = angular.element(
+            document.getElementById('questionSuggestionDiv'));
+          questionSuggestionDiv.on('scroll', function() {
+            if (questionSuggestionDiv.scrollTop() > STICKY_HEADER_SCROLL_TOP) {
+              $scope.isSticky = true;
+            } else {
+              $scope.isSticky = false;
+            }
+            $scope.$applyAsync();
+          });
```

Comment:
Can't we do a CSS only solution like [here](https://css-tricks.com/creating-sliding-effects-using-sticky-positioning/)?

Reply:
A closer look would reveal that I have done the same thing here. But I did not stop at that, because as the Overview of this PR describes "...the width of the sticky div containing the question would reduce the visibility of the contents below it. Thus, a function was written to reduce the height of the div when it was sticky. This allowed for looking at the question without scrolling while increasing the visibility of the scrolled region."

To rephrase, making it sticky reduced the scrollable view. So, I reduced the height of the sticky div after it became sticky. That needed a trigger when the header became sticky. So we wrote a function. Reducing that way did not provide enough visibility, so we reduced the margins.

Reply:
This is inefficient IMO -- triggering a `$applyAsync` everytime the scroll event is triggered.

More generally, keep styling related changes only in the template and not in the corresponding `ts` files. We avoid this in most of our directives & components. Have you evaluated any other solutions?

Reply:
Okay. I have discarded the `$applyAsync()` function. I, too, hold the same view that styling-related changes must be in the template and not in the corresponding ts files. But I think you have missed a piece of the solution here since you seem to be under the impression that this function is implementing the sticky behavior of the header. Please go through the details of the implementation I have posted below.